app/src/main/java/org/fossasia/susi/ai/adapters/viewholders/LinkPreviewViewHolder.java

```diff
@@ -43,7 +72,179 @@
+                        } else {
+                            previewImageView.setVisibility(View.VISIBLE);
+                            Picasso.with(currContext).load(imageList.get(0))
+                                    .fit().centerCrop()
+                                    .into(previewImageView);
+                            link.setImageURL(imageList.get(0));
+                        }
+                    }
+
+                    model.setWebLinkData(link);
+                    realm.copyToRealmOrUpdate(model);
+                    realm.commitTransaction();
+                }
+            };
+
+            if (model != null) {
+                List<String> urlList = ((MainActivity)currContext).recyclerAdapter.extractLinks(model.getContent());
+                StringBuilder url = new StringBuilder(urlList.get(0));
+                StringBuilder http = new StringBuilder("http://");
+                StringBuilder https = new StringBuilder("https://");
+                if (!(url.toString().startsWith(http.toString()) || url.toString().startsWith(https.toString()))) {
+                    url = http.append(url.toString());
+                }
+                TextCrawler textCrawler = new TextCrawler();
+                textCrawler.makePreview(linkPreviewCallback, url.toString());
+            }
+        } else {
+
+            if(!model.getWebLinkData().getHeadline().isEmpty()) {
+                Log.d(TAG, "onPos: " + model.getWebLinkData().getHeadline());
+                titleTextView.setText(model.getWebLinkData().getHeadline());
+            } else {
+                titleTextView.setVisibility(View.GONE);
+                Log.d(TAG, "handleItemEvents: " + "isEmpty");
+            }
+
+            if(!model.getWebLinkData().getBody().isEmpty()) {
+                Log.d(TAG, "onPos: " + model.getWebLinkData().getHeadline());
+                descriptionTextView.setText(model.getWebLinkData().getBody());
+            } else {
+                descriptionTextView.setVisibility(View.GONE);
+                Log.d(TAG, "handleItemEvents: " + "isEmpty");
+            }
+
+            if(model.getWebLinkData().getHeadline().isEmpty() && model.getWebLinkData().getBody().isEmpty()) {
+                previewLayout.setVisibility(View.GONE);
+            }
+
+            Log.i(TAG, model.getWebLinkData().getImageURL());
+            if (!model.getWebLinkData().getImageURL().equals("")) {
+                Picasso.with(currContext).load(model.getWebLinkData().getImageURL())
+                        .fit().centerCrop()
+                        .into(previewImageView);
+            } else {
+                previewImageView.setVisibility(View.GONE);
+            }
+
+            url = model.getWebLinkData().getUrl();
+        }
+
+        text.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (((MainActivity)currContext).recyclerAdapter.selectedItems.size() != 0)
```

Comment:
Remove this, pass adapter as param and use `adapter.selectedItems.size()` Do it everywhere.